std/encoding.d

```diff
@@ -2462,7 +2462,7 @@ abstract class EncodingScheme
      * Params:
      *    s = the array to be tested
      */
-    bool isValid(const(ubyte)[] s)
+    bool isValid()(const(ubyte)[] s)
```

Comment:
Not that the overriden method doesn't have to match exactly the attributes, only to provide a covariant overload.

So it is totally legitimate to override `void foo () nothrow` with `void foo () nothrow @safe` because you extend the promises the interface is making.

So one possible approach (not tested, not investigated, just some food for though) would be to go with template functions which uses the actual type of the implementation rather than the interface, so one can define an interface that has all the attributes and base its encoding on that.
However that is probably much more work and design, especially if you didn't use OOP much in D.